src/app/ReadHandler.h

```diff
@@ -170,6 +170,11 @@ class ReadHandler : public Messaging::ExchangeDelegate
         /// @param[in] apReadHandler  ReadHandler getting added
         virtual void OnReadHandlerCreated(ReadHandler * apReadHandler) = 0;
 
+        /// @brief Callback invoked when a ReadHandler's reporting intervals is changed through SetMaxReportingInterval or
+        /// SetMinReportingIntervalForTests
+        /// @param[in] apReadHandler  ReadHandler that change its read intervals
+        virtual void OnReportingIntervalsChanged(ReadHandler * apReadHandler) = 0;
```

Comment:
Since in normal operations you can't change the reporting intervals, this should only be compiled when building the associated tests.

Reply:
This seems to be used for tests only so it would need to be `ifdef` in only for tests.

Reply:
This callback was removed as suggested by @bzbarsky-apple